sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -119,6 +124,54 @@ pub struct SledConfig {
     pub zones: Vec<BlueprintZoneConfig>,
```

Comment:
Struggling to comment the right line here -- I'm trying to comment on the `pub datasets: DatasetsConfig` line above -- but basically:

- We added this field in https://github.com/oxidecomputer/omicron/pull/6229
- but we never populated it correctly

That's what this PR is most significantly trying to fix